Objects/Converters/ConverterRevit/ConverterRevitShared/Partial Classes/Units.cs

```diff
@@ -202,9 +203,20 @@ public static double ScaleToSpeckleStatic(double value, ForgeTypeId forgeTypeId)
 
     public static double ScaleToSpeckle(double value, ForgeTypeId forgeTypeId, IRevitDocumentAggregateCache cache)
     {
-      return value * cache
-        .GetOrInitializeEmptyCacheOfType<double>(out _)
-        .GetOrAdd(forgeTypeId.TypeId, () => UnitUtils.ConvertFromInternalUnits(1, forgeTypeId), out _);
+      if (forgeTypeId.IsLengthType())
+      {
+        return value * cache
+          .GetOrInitializeEmptyCacheOfType<double>(out _)
+          .GetOrAdd(forgeTypeId.TypeId, () => UnitUtils.ConvertFromInternalUnits(1, forgeTypeId), out _);
+      }
+      else
+      {
+        var cacheKey = $"{forgeTypeId.TypeId}_{value}";
```

Comment:
Could you benchmark the halfway house vs. not implementing a cache at all?

It's a shame that there isn't a check enum on the `forgeTypeId` that says whether ConvertFromInternals is linear. I was partly against the building yet another area where we have conditionals based on Revit internals - even more so when it relates to a black-box API call results.